wharehauora-wifi-gateway/wharehauora-wifi-gateway.ino

```diff
@@ -23,17 +49,56 @@ char mqtt_password[32];
 #define MY_ESP8266_SSID my_ssid
 #define MY_ESP8266_PASSWORD my_pass
 #define MY_CONTROLLER_URL_ADDRESS my_server
-#define MY_MQTT_PUBLISH_TOPIC_PREFIX "/sensors/wharehauora/5"
-#define MY_MQTT_SUBSCRIBE_TOPIC_PREFIX "/sensors/wharehauora/5"
+#define MY_MQTT_PUBLISH_TOPIC_PREFIX mqtt_publish_topic
+#define MY_MQTT_SUBSCRIBE_TOPIC_PREFIX "/sensors/wharehauora/"
+
+
 #define MY_MQTT_USER mqtt_username
 #define MY_MQTT_PASSWORD mqtt_password
-#define MY_PORT 16259 //not-ssl
-//#define MY_PORT 26259 /// SSL
+
 
 #include <MySensors.h>
 
 bool shouldSaveConfig = false;
 
+/**
+ * save custom parameters that wifimanager collects to EEPROM
+ */
+void saveCustomParameters(){
+  for(int i = 0; i < 32; i++){
+    saveState(i + WHARE_USERNAME_POSITION, mqtt_username[i]);
+  }
+
+  for(int i = 0; i < 32; i++){
+    saveState(i + WHARE_PASSWORD_POSITION, mqtt_password[i]);
+  }
+}
+
+/**
+ * Here is where we attempt to retrieve the whare_mqtt_username and whare_mqtt_password from the eeprom
+ *
+ * How do I know that what I am reading is gobbledygook or
+ * legitimate data?
+ * Does it even matter?
+ *
+ * how to we reset them if they are wrong but the user has got the other stuff correct?
+ */
+void loadCustomParameters(){
+  Serial.println("loading custom parameters");
```

Comment:
learned that `Serial.println` is used to see the data being generated by the Arduino.

Reply:
Yes, it's debug. We should probably wrap it in `#ifdef DEBUG` soon, so we don't use up memory and time printing debug we're not in development mode.

Reply:
Yeah I was just thinking that this handy when deving but useless in the wild....